src/librustdoc/html/render/mod.rs

```diff
@@ -2829,7 +2829,21 @@ fn render_call_locations<W: fmt::Write>(
                 file.start_pos + BytePos(byte_max),
             ))
         })()
-        .unwrap_or(DUMMY_SP);
+        .unwrap_or_else(|| {
```

Comment:
Instead of cloning parts of the above code, I think it'd might be simpler to instead tweak it a bit:

```rust
            let local = tcx.sess.local_crate_source_file()?;
            let crate_src = local.into_local_path()?;
            let abs_crate_src = crate_src.canonicalize().ok()?;
            let crate_root = abs_crate_src.parent()?.parent()?;
            let rel_path = path.strip_prefix(crate_root).ok()?;
            let files = source_map.files();
            let file = files.iter().find(|file| match &file.name {
                FileName::Real(RealFileName::LocalPath(other_path)) => rel_path == other_path,
                _ => false,
            }).or_else(|| files.iter().find(|file| matches!(&file.name, FileName::Real(file_name) if file_name == &local))?;
```

Reply:
Then we can replace the `unwrap_or(DUMMY_SP)` with `expect("no local file found")`.